Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -8390,6 +8391,29 @@ void WebPage::updateLookalikeCharacterStringsIfNeeded()
 
 #endif // ENABLE(NETWORK_CONNECTION_INTEGRITY)
 
+bool WebPage::shouldSkipDecidePolicyForResponse(const WebCore::ResourceResponse& response, const WebCore::ResourceRequest& request) const
+{
+    if (!m_skipDecidePolicyForResponseIfPossible)
+        return false;
+
+    constexpr auto noContent = 204;
+    constexpr auto smallestError = 400;
+    auto statusCode = response.httpStatusCode();
+    if (statusCode == noContent || statusCode >= smallestError)
+        return false;
+
+    if (!equalIgnoringASCIICase(response.mimeType(), "text/html"_s))
+        return false;
+
+    if (response.url().isLocalFile())
+        return false;
+
+    if (auto components = response.httpHeaderField(HTTPHeaderName::ContentDisposition).split(';'); !components.isEmpty() && equalIgnoringASCIICase(stripLeadingAndTrailingHTTPSpaces(components[0]), "attachment"_s))
+        return false;
```

Comment:
Is the idea here that WebKit now short-circuits decide policy without needing to call out to the delegate when it knows that Safari's delegate would not doing anything interesting?

I think that's a fine approach, especially as a stopgap to resolve a regression. But it probably needs a name that better conveys the conditions of "if possible". Maybe "shouldSkipDecidePolicyForResponseInBrowserFastPaths", or something like that.